hack/tools/release/notes/test/golden/v1.5.0.md

```diff
@@ -161,48 +161,36 @@ REPLACE ME: A couple sentences describing the deprecation, including links to do
 - clusterctl: Update cert-manager to v1.12.2 (#8883)
 - Core: Cache unstructured in Cluster, MD and MS controller (#8916)
 - Core: Remove unnecessary requeues (#8743)
-- Dependency: Bump actions/checkout from 3.3.0 to 3.4.0 (#8321)
-- Dependency: Bump actions/checkout from 3.4.0 to 3.5.0 (#8389)
-- Dependency: Bump actions/checkout from 3.5.0 to 3.5.2 (#8540)
-- Dependency: Bump actions/checkout from 3.5.2 to 3.5.3 (#8837)
-- Dependency: Bump actions/setup-go from 3.5.0 to 4.0.1 (#8664)
+- Dependency: Bump cert-manager to v1.13.0 (#9413)
+- Dependency: Bump cert-manager to v1.13.1 (#9507)
+- Dependency: Bump cert-manager to v1.13.2 (#9658)
+- Dependency: Bump controller runtime to v1.15.3 (#9624)
+- Dependency: Bump controller-runtime to v0.15.1 (#9127)
+- Dependency: Bump corefile-migration library to v1.0.21 (#9309)
```

Comment:
Looks like the tool is listing merged PRs up to the date of the [latest commit on the release branch](https://github.com/willie-yao/cluster-api/blob/release-notes-dependencies/hack/tools/release/notes/list.go#L72). It looks like there are PRs still being cherry-picked into v1.5.0 which is why these PRs show up. Is this something we should worry about? In a normal release cycle, we'll be generating the release notes at the correct day, but if we retroactively generate them, we might be getting some new commits that aren't part of the release.

I'm thinking maybe we can add a flag for a specific end date, or we can just manually remove the newer PRs since I don't think we'd be retroactively generating release notes in the normal release flow.

Reply:
Umm I see
I suspect there is bug in how the tests are setup then
Because they are supposed to use "static" config so the output doesn't change even if we merge more commits in the original branch.
We might be missing a `to` ref or the ref might be incorrect

Reply:
I used the command `RELEASE_TAG=v1.5.0 make release-notes` to generate the golden files so I don't think it's just a bug in the test. I think it looks for the latest commit in the branch rather than when the tag is created: https://github.com/willie-yao/cluster-api/blob/release-notes-dependencies/hack/tools/release/notes/list.go#L65

Reply:
I see that it is being defaulted to `heads/branch` instead of `tags/tagname` [here](https://github.com/willie-yao/cluster-api/blob/release-notes-dependencies/hack/tools/release/notes/main.go#L183). Should we make `tags/` the default if `RELEASE_TAG` is specified? I don't see why a user would need to read from `heads/branch` if they are generating release notes for a specific release tag.